main/modes/games/cGrove/Garden/cg_Grove.c

```diff
@@ -0,0 +1,670 @@
+// Defines
+//==============================================================================
+
+#define CG_CURSOR_SPEED 16
+
+//==============================================================================
+// Consts
+//==============================================================================
+
+static const char* groveCursorSprites[] = {
+    "chowa_hand1.wsg",
+    "chowa_hand2.wsg",
+    "chowa_hand3.wsg",
+};
+
+static const char* questionMarkSprites[] = {
+    "questmk-1.wsg", "questmk-2.wsg", "questmk-3.wsg", "questmk-4.wsg", "questmk-5.wsg", "questmk-6.wsg",
+};
+
+static const char* angerParticles[] = {
+    "anger-1.wsg",
+    "anger-2.wsg",
+};
+
+static const char* musicNoteSprites[] = {
+    "cg_note1.wsg",
+    "cg_note2.wsg",
+    "cg_note3.wsg",
+};
+
+static const char* speechBubbleSprites[] = {
+    "cg_text0.wsg",
+    "cg_text1.wsg",
+    "cg_text2.wsg",
+    "cg_text3.wsg",
+};
+
+static const char* itemSprites[] = {
+    "agi_book.wsg",   "cha_book.wsg", "spd_book.wsg",     "sta_book.wsg", "str_book.wsg", "cg_ball.wsg",
+    "cg_crayons.wsg", "cg_knife.wsg", "cg_toy_sword.wsg", "cake.wsg",     "souffle.wsg",  "DonutRing.wsg",
+};
+
+static const char shopMenuTitle[] = "Ring Shop";
+
+static const char* shopMenuItems[] = {
+    "Agility Stat Book",
+    "Charisma Stat Book",
+    "Strength Stat Book",
+    "Stamina Stat Book",
+    "Speed Stat Book",
+    "Souffle",
+    "Ball",
+    "Crayons",
+    "Toy Sword",
+    "Back",
+};
+
+//==============================================================================
+// Function Declarations
+//==============================================================================
+
+/**
+ * @brief Attempts to grab objects. Due to total amount being limited, no need to optimize
+ *
+ * @param cg Game Object
+ */
+static void cg_attemptGrab(cGrove_t* cg);
+
+/**
+ * @brief Input handling for garden
+ *
+ * @param cg Game Object
+ */
+static void cg_handleInputGarden(cGrove_t* cg);
+
+/**
+ * @brief Moves the view of the field byt eh provided x and y
+ *
+ * @param cg Game Object
+ * @param xChange Distance to move horizontally. Negative is right, positive if left.
+ * @param yChange Distance to move vertically. Negative is up, positive s down
+ */
+static void cg_moveCamera(cGrove_t* cg, int16_t xChange, int16_t yChange);
+
+/**
+ * @brief Initialize object boundaries
+ *
+ * @param cg Game Data
+ */
+static void cg_setupBorders(cGrove_t* cg);
+
+/**
+ * @brief Menu callback
+ *
+ * @param label Selection labels
+ * @param selected If selected
+ * @param settingVal Value of a changed setting
+ */
+static void shopMenuCb(const char* label, bool selected, uint32_t settingVal);
+
+/**
+ * @brief Callback to restart BGM
+ *
+ */
+static void cg_bgmCB(void);
+
+//==============================================================================
+// Variables
+//==============================================================================
+
+bool isBGMPlaying;
```

Comment:
Not really. `cgr` is just a copy of the `cg` game object used because you can't pass the main game object to the menu callback. the main cg data object does not control Grove mechanics.

That said, it should probably be in cg->grove, which is the grove data handler.

Reply:
You know what I meant

Reply:
lol